raspi-config

```diff
@@ -2611,21 +2612,21 @@ EOF
   update-initramfs -c -k "$KERN"
 
   # rename it so we know it has overlay added
-  mv /boot/initrd.img-"$KERN" /boot/"$INITRD"
+  mv /boot/initrd.img-"$KERN" "$BOOT"/"$INITRD"
```

Comment:
Sorry about the delay

I'm guessing the /boot on this particular line is expected because update-initramfs will always write to /boot?

Reply:
btw: When patching cmdline.txt (after init_resize) scripts should really check the overlay_prefix parameter from /proc/device-tree/chosen  should be checked.

https://www.raspberrypi.com/documentation/computers/configuration.html#part4

Otherwise, it's possible that the wrong cmdline.txt will be patched

Reply:
> check the overlay_prefix parameter from /proc/device-tree/chosen should be checked.

`overlay_prefix` or `os_prefix`?

Also, wouldn't the whole mechanism be broken on Pi OS, which assumes that /boot is THE place everything is?

In either case, I think that one is for @spl237, since the overlayfs parts of raspi-config are his.

Reply:
OK, I've now spent ten minutes scratching my head about this...

What does the overlay_prefix parameter from /proc/device-tree (which is used for loading kernel overlays) have to do with patching cmdline.txt (which is used for adding a file system overlay)? Are those things not completely different, with the only common factor being the word "overlay"? Or am I missing something?

Reply:
Sorry, I mean os_prefix

Reply:
Thanks! Now fixed.

Reply:
> Thanks! Now fixed.

I'm pretty sure it only breaks things. All the related packages install files to /boot and it's hardcoded all over the place (which is reasonable).

If things are instead in a subdirectory, nothing is going to work. The system should be set up to bind /boot to the appropriate subdirectory instead.

Reply:
Well, it is now in a global variable in raspi-config which is used instead of /boot/cmdline.txt throughout. So if it does break things, it's trivial to remove - just hard-code PREFIX to an empty string on line 24.

I've got no idea what this breaks or doesn't break - I just do what I'm told... ;)

Reply:
Had a chat with Tim and yeah, it breaks things, but not in the context `os_prefix` is expected to be used in. I'll move it to public and get this PR sorted when I've got a spare moment.

Thanks